### DISCUSSION TITLE
Can Binding Directly To Model Properties work after extracting a form object

### DISCUSSION THREAD

**Discussion by raliqala** [livewire/livewire#8001] [1. Help] [answered]
Good day

I'm trying to bind my form to model properties after extracting a form object. This due to the security suggestions in the documentation.
But I don't know if binding directly to model properties will be as secure as when extracting a form object or if these 2 are the same? https://livewire.laravel.com/docs/security#using-model-properties

Thanks in Advance.

**Comment by joshhanley** [comment]
@raliqala I'm not exactly sure what you mean. Can you share some code of what you are trying to do?

**Comment by raliqala** [comment]
This is the extracted form object from my users component.  I want to bind the front-end form directly to my model properties, but I know when I use extracted form object I'm going to bind my front-end form like : 
 ```HTML 
 public UserForm $form; 
<input type="text" wire:model="{Form variable}.email">
```  
But if I use the code in my UserForm below will I be able to do something like: 
```PHP
<?php
 
namespace App\Livewire;
 
use App\Livewire\Forms\PostForm;
use Livewire\Component;
use App\Models\Post;
 
class CreatePost extends Component
{
    public UserForm $form; 
 
    public function save()
    {
        $this->validate();
 
        Post::create(
            $this->form->all() 
        );
 
        return $this->redirect('/users');
    }
 
    public function render()
    {
        return view('livewire.create-user');
    }
}
<input type="text" wire:model="form.email"> //? even if I'm binding to my model
```

![image](https://github.com/livewire/livewire/assets/30502407/e1f8bbee-80c0-4f53-8985-12a057cfd014)  

My question is for security purposes "is binding my front-end form to my model or binding my front-end form to the UserForm the same? or i can just do as i did in the above image and everything will be fine it won't be a repeating code"

I'm doing this https://livewire.laravel.com/docs/forms/#extracting-a-form-object but going the extra mile of using my model instead of declaring public variables.

I don't know if this is clear, please bear with me

**Reply by joshhanley** [reply, marked as answer]
@raliqala ah right, so we don't recommend using model binding, in fact it is disabled in V3 by default. The purpose of the form object is that the properties on it should match your form fields directly (regardless of the model).

You can then use the `$form->validate()` to get validated data out of the form in an array which you can use to populate your model. For example `Post::create($this->form->validate());`
